src/System.Runtime.Analyzers/Core/NormalizeStringsToUppercase.cs

```diff
@@ -4,45 +4,98 @@
-                                                                             DiagnosticCategory.Naming,
-                                                                             DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
-                                                                             description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
-                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
         internal static DiagnosticDescriptor ToUpperRule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessageToUpper,
-                                                                             DiagnosticCategory.Naming,
+                                                                             DiagnosticCategory.Globalization,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: @"https://msdn.microsoft.com/en-us/library/bb386042.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(ToUpperInvariantRule, ToUpperRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(ToUpperRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterCompilationStartAction(compilationStartContext =>
+            {
+                var stringType = WellKnownTypes.String(compilationStartContext.Compilation);
+                if (stringType == null)
+                {
+                    return;
+                }
+
+                var cultureInfo = compilationStartContext.Compilation.GetTypeByMetadataName("System.Globalization.CultureInfo");
+                var invariantCulture = cultureInfo?.GetMembers("InvariantCulture").OfType<IPropertySymbol>().FirstOrDefault();
```

Comment:
If `toLowerWithCultureInfo` is null (i.e. the compilation has no method `string.ToLower(CultureInfo)`), how can the user code invoke that method?

Basically, the portable profile doesn't have `string.ToLower(CultureInfo)`, and we want to still flag calls to `string.ToLowerInvariant()` and bail out on the rest. Do you see a case where this is not handled correctly here?